### PR TITLE
Always display max upload size in admin panel

### PR DESCRIPTION
**Pull request by rullzer**
Patch for #10600 (and by extend #523).

Always show the max upload size. But just disable it if we cannot change it. As stated in #10600 we probably should add some extra text to explain why it is greyed out... Any designers with ideas?

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Nice initiative - THX

**Comment by DeepDiver1975** [comment]
@jancborchardt can you please have a look at the UX? THX

**Comment by karlitschek** [comment]
:+1:  from my side

**Comment by DeepDiver1975** [comment]
tested :+1:

**Comment by DeepDiver1975** [comment]
@jancborchardt 

bildschirmfoto von 2015-02-13 10 51 45

**Comment by jancborchardt** [comment]
If it can't be edited, we should replace the "Save" button with an explanatory text like:

> Can not be edited from here due to insufficient permissions.

Otherwise it's not clear why it's greyed out.

Does that sound good/correct @Deepdiver1975?

**Comment by DeepDiver1975** [comment]
> Does that sound good/correct @Deepdiver1975?

agreed - @rullzer can I ask you to update your PR - Thanks a lot!

**Comment by rullzer** [comment]
Done

**Comment by DeepDiver1975** [comment]
@jancborchardt 

bildschirmfoto von 2015-02-13 12 22 16

**Comment by PVince81** [comment]
Should we mention `.htaccess` here ? Insufficient permissions sounds a bit vague.

Also I believe that "can not" should be "cannot"

**Comment by rullzer** [comment]
@PVince81 we could mention `.haccess` but we should keep it generic since the user does not have to run apache of course.

Something like: `Cannot edit .htaccess to modify maximum upload size.`

However that might make the user believe that editing the .htaccess file will make everything all right. We could also check `$_SERVER['SERVER_SOFTWARE']` or `apache_get_version`. To only mention something about `.htaccess` if the server is apache and something like:

`To edit the maximum upload size on your server please consult your servers documentation.`

in the other cases.

File: apps/files/templates/admin.php [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 		<?php endif;?>
 		<br/>
 		<input type="hidden" value="<?php p($_['requesttoken']); ?>" name="requesttoken" />
-		<input type="submit" name="submitFilesAdminSettings" id="submitFilesAdminSettings"
-			   value="<?php p($l->t( 'Save' )); ?>"/>
+		<?php if($_['uploadChangable']): ?>
+			<input type="submit" name="submitFilesAdminSettings" id="submitFilesAdminSettings"
+				   value="<?php p($l->t( 'Save' )); ?>"/>
+		<?php else: ?>
+			<?php p($l->t('Can not be edited from here due to insufficient permissions.')); ?>
```
Comment (nickvergessen) [inline apps/files/templates/admin.php]: Should we state what exactly needs to be changed where?

**Comment by nickvergessen** [comment]
Change looks good in general. 

The message is displayed when
`!(getenv('htaccessWorking') == 'true' && is_writable(OC::$SERVERROOT.'/.htaccess'))`

So we could display something like `because your .htaccess is not writeable or not used at all.` ?

**Comment by MorrisJobke** [comment]
Please don't overengineer this. I would simply add a reference to the manual and there you can list the different webserver setups.

Tested looks good so far :+1:

**Comment by DeepDiver1975** [comment]
> Please don't overengineer this. I would simply add a reference to the manual and there you can list the different webserver setups.

:+1:

**Comment by jospoortvliet** [comment]
This setting still does get lost on update/upgrade, right?

**Comment by LukasReschke** [comment]
Yes. If it gets written to the .htaccess it gets overwritten.

**Comment by jospoortvliet** [comment]
ok, tnx.

**Pull request merged**